Adatb1/2zh/feladat7_plsql_megoldasok.txt

```diff
@@ -0,0 +1,115 @@
+set serveroutput on
+
+CREATE OR REPLACE
+FUNCTION prim(n integer) RETURN number IS
+  m integer := 2;
+  root integer := sqrt(n);
+BEGIN
+  while m < root loop
+    if mod(n, m) = 0 then
+      return (0);
+    end if;
+    m := m + 1;
+  end loop;
+
+  return (1);
+END;
+/
+SELECT prim(26388279066623) from dual;
+
+
+CREATE OR REPLACE
+PROCEDURE fib(n integer) IS
+  p1 integer := 0;
+  p2 integer := 1;
+  sums integer;
+begin
+  for i in 2..n loop
+    sums := p1 + p2;
+    p1 := p2;
+    p2 := sums;
+  end loop;
+  dbms_output.put_line(p1);
+end;
+/
+set serveroutput on
+call fib(10);
+
+
+CREATE OR REPLACE
+FUNCTION lnko(p1 integer, p2 integer) RETURN number IS
+  a integer := p1;
+  b integer := p2;
+begin
+  while a != b loop
+    if a > b then
+      a := a - b;
+    else
+      b := b - a;
+    end if;
+  end loop;
+  return a;
+end;
+/
+SELECT lnko(3570,7293) FROM dual;
+
+
+CREATE OR REPLACE
+FUNCTION faktor(n integer) RETURN integer IS
+  fac integer := 1;
+begin
+  for i in 1..n loop
```

Comment:
For example:
CREATE OR REPLACE FUNCTION faktor(n integer) RETURN integer IS
BEGIN
  if (n=0) or (n=1) then
    return 1;
  else
    return faktor(n)*faktor(n-1);
  end if;
END;
/
SELECT faktor(10) FROM dual;